src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -981,7 +1015,10 @@ impl MooncakeTable {
         if !self.next_snapshot_task.should_create_snapshot() && !opt.force_create {
             return false;
         }
+        let start = Instant::now();
         self.create_snapshot_impl(opt);
+        let elapsed = start.elapsed().as_secs_f64();
+        self.snapshot_stats.update(elapsed);
```

Comment:
I would recommend to make the metrics collect function to accept a closure, so the code can be like:
```suggestion
        self.snapshot_stats.record_create_snapshot_duration(
          self.create_snapshot_impl(opt);
        );
```

but I'm not sure what is the best syntax to do this in rust, so please take this a grain of salt :)